content/blog/2023/11/2023-11-29-jenkins-to-github.adoc

```diff
@@ -0,0 +1,269 @@
+With them, you can write long essays of messages to the "Checks" tab of a PR, but they do require authenticating as a GitHub App instead of a user, so it's more complicated to set up.
+SonarCloud, a code analyzer, makes use of PR Checks via https://github.com/apps/sonarcloud[their namesake GitHub App].
+If you look closer, their report is just Markdown:
+image:/images/post-images/2023/11/29/20230902151934-Screenshot-of-a-SonarCloud-Code-Analysis-as-a-GitHub-PR-Check.png[Screenshot of a SonarCloud Code Analysis as a GitHub PR Check]
+
+. *GitHub Actions* (GA).
+GA is a different kind of CI system from Jenkins.
+Since this article is about Jenkins only, GA is only mentioned here for the sake of completeness.
+image:/images/post-images/2023/11/29/20230902152320-Screenshot-of-a-GitHub-Action-as-a-PR-Check.png[Screenshot of a GitHub Action as a PR Check]
+
+== When the result is simple: Use commit statuses
+
+The Jenkins jobs you want to trigger only runs for a couple of seconds.
+Each only emits a simple result: pass or fail.
+You don't need fancy reports, so *commit statuses* seem to be a suitable choice.
+
+You installed https://plugins.jenkins.io/github/[the GitHub plugin] to your Jenkins server and followed https://stackoverflow.com/a/51003334/1147061[this guide] to set it up.
+You are able to emit simple one-liners as checks now, like this:
+
+image:https://raw.githubusercontent.com/jenkinsci/github-coverage-reporter/readme/assets/coverage-success.png[]
+_(https://github.com/jenkinsci/github-coverage-reporter-plugin[source] of screenshot; Apache 2.0 licensed)_
+
+It met your immediate need, but it stills falls short in a number of ways:
+
+* The plugin publishes to PR only thrice: upon getting queued, upon the start, and upon completion.
+Your job actually has *many time-consuming stages* (e.g., building a manifest, deploying to staging, and running tests), often packed in a single script, and you *want to be notified at each stage*.
+* The messages show up with *your GitHub profile image as the icon*, because Jenkins is using _your_ PAT!
+You want to shy away from the spotlight, maybe because you don't want to be mistaken by your less-GitHub-savvy collaborators as a ruthless robot critiquing (and blocking) everybody's PRs.
+
+That's when you start to look further.
+
+== When the report is long: Use PR checks
+
+Your job generates a long report at the end, but the Checks section can only accommodate a couple of words.
+You have been rendering the reports as HTML and attaching them with the Jenkins Builds themselves (perhaps with a nice https://palletsprojects.com/p/jinja/[Jinja2 template] and https://plugins.jenkins.io/htmlpublisher/[the HTML Publisher plugin]).
+The commit statuses your job writes to PRs contained links to the Jenkins Builds themselves, so it's only two clicks away from an HTML report -- not bad.
+
+To prevent past HTML reports from saturating the disk, you configured the job to only keep the 20 most recent Builds.
```

Comment:
```suggestion
To prevent past HTML reports from saturating the disk, you configured the job to only keep the 20 most recent builds.
```
Builds does not need to be capitalized unless it is beginning a sentence.